pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -38,11 +37,21 @@ class TrainingTypePlugin(Plugin, ABC):
     Base class for all training type plugins that change the behaviour of the training, validation and test-loop.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, checkpoint_plugin: Optional[CheckpointIOPlugin] = None) -> None:
         self._model: Optional[Module] = None
         self._results: Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]] = None
+        checkpoint_plugin = checkpoint_plugin if checkpoint_plugin is not None else TorchCheckpointIOPlugin()
+        self._checkpoint_plugin: CheckpointIOPlugin = checkpoint_plugin
```

Comment:
```suggestion
        self._checkpoint_plugin = checkpoint_plugin
```

I don't think this annotation is necessary, mypy should be able to figure it out 🤔